### PR TITLE
Add debug AABB visualization and fix to tile bounds debug visualization during globe transition

### PR DESCRIPTION
**Pull request by SnailBones**
Closes https://github.com/mapbox/gl-js-team/issues/472

Also fixes an issue with tile bounds visualization during the transition not being correctly scaled, as demonstrated by changes in render tests.

image

The AABBs can be viewed from other angles with the `transform.freezeTileCoverage` option

image

 - [ ] Implement clipping to fix artifacts
 - [X] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [X] manually test the debug page
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [X] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

File: src/render/draw_debug.js [hunk 1]
```diff
@@ -210,3 +215,96 @@ function drawTextToOverlay(painter: Painter, text: string) {
...
+export function drawAabbs(painter: Painter, sourceCache: SourceCache, coords: Array<OverscaledTileID>) {
+    const tr = painter.transform;
+
+    const worldToECEFMatrix = mat4.invert(new Float64Array(16), tr.globeMatrix);
+    const ecefToPixelMatrix = mat4.multiply(mat4.identity(new Float64Array(16)), tr.pixelMatrix, tr.globeMatrix);
+    const numTiles =  1 << tr.coveringZoomLevel({tileSize: tr.tileSize});
+
+    if (!tr.freezeTileCoverage) {
+        aabbCorners = coords.map(coord => {
+            const aabb = aabbForTileOnGlobe(tr, numTiles, coord.canonical);
```
Comment (akoylasar) [inline src/render/draw_debug.js:278]: instead of `numTiles` we could also pass in `tr.worldSize` and then in the for-loop avoid the scaling.

**Comment by akoylasar** [comment]
looks nice and it's very useful!

**Review by akoylasar — APPROVED** [review]

File: src/render/draw_debug.js (hunk 1 above)
Comment (SnailBones) [reply src/render/draw_debug.js:278]: Not sure I understand, are you suggesting reducing the number of arguments passed into `aabbForTileOnGlobe`? It's called in one other place [here](https://github.com/mapbox/mapbox-gl-js/blob/main/src/geo/projection/tile_transform.js#L99) so we'd need to make the same changes there.

Comment (akoylasar) [reply src/render/draw_debug.js:278]: I meant to say we could pass in `tr.worldSize` instead of `numTiles`.

Comment (SnailBones) [reply src/render/draw_debug.js:278]: Are you suggesting that we use `tr.worldSize` to calculate numTiles inside of `aabbForTileOnGlobe`?

**Review by mourner — COMMENTED** [review]
My only concern here is that this very specialized tool adds ~700 bytes overhead to the gzipped bundle for all users, while it's a feature that would only be used by us, the developers. Could we perhaps limit it to the dev bundle and strip it out like we do with a few other debug code pieces?

**Comment by SnailBones** [comment]
> My only concern here is that this very specialized tool adds ~700 bytes overhead to the gzipped bundle for all users, while it's a feature that would only be used by us, the developers. Could we perhaps limit it to the dev bundle and strip it out like we do with a few other debug code pieces?

Good call @mourner! I just implemented this, bringing the change down from +683 B to +88 B.

**Review by mourner — APPROVED** [review]

**Pull request merged**